### ISSUE TITLE
Undefined properties of requests are converted to 0 in the callback of createService

### ISSUE THREAD

**Issue by qiuzhong** [RobotWebTools/rclnodejs#243]
Let's say the client sends an ill-formatted request `{a:1}` for the service `example_interfaces/srv/AddTwoInts` (which requires two properties in the request), in the callback of 
`createService`:
```
  node.createService('example_interfaces/srv/AddTwoInts', 'add_two_ints', (request, response) => {
    console.log(`Incoming request: ${typeof request}`, request);
    let result = response.template;
    result.sum = request.a + request.b;
    console.log(`Sending response: ${typeof result}`, result, '\n--');
    response.send(result);
  });
```
We get this output:
```
Incoming request: object { a: 69, b: 0 }
Sending response: object { sum: 69 } 
--

```

Undefined property `request.b` is converted to 0 and doesn't throw an Error. If you print the type of request.b, it's `number`.

**Comment by qiuzhong** [comment]
@kenny-y , @minggangw , what do you think?

**Comment by minggangw** [comment]
Sorry for the late response, when we create a JavaScript object in `rclnodejs`, and a c struct memory will be allocated simultaneously, which is done by [ref](https://www.npmjs.com/package/ref). So, I think why you get `b == 0` is that the `ref` sets default value for the members of the struct, even though there is no corresponding property in JavaScript object.

**Comment by qiuzhong** [comment]
So is this the well-defined behavior for `rclnodejs`? If so, I don't think it's a robust service to accept ill-formatted requests.

**Comment by kenny-y** [comment]
I think this and #241 can fall into the same category -- strict structure/type checking before using the object

**Comment by qiuzhong** [comment]
Fixed by #274
